src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -311,212 +334,219 @@ void func_80BA8DF4(EnBaba* this, GlobalContext* globalCtx) {
         Math_SmoothStepToS(&this->torsoRot.y, 0, 4, 0x3E8, 1);
     }
 
-    SubS_FillLimbRotTables(globalCtx, this->unk_302, this->unk_326, ARRAY_COUNT(this->unk_302));
+    SubS_FillLimbRotTables(globalCtx, this->limbRotTableY, this->limbRotTableZ, ARRAY_COUNT(this->limbRotTableY));
 
-    if (this->unk_40A & 2) {
-        func_80BA8C90(this, globalCtx);
+    if (this->flags & ENBABA_VISIBLE) {
+        EnBaba_UpdateCollider(this, globalCtx);
     }
 }
 
-s32 func_80BA8F88(EnBaba* this, GlobalContext* globalCtx, ScheduleResult* arg2) {
-    u16 sp26 = (u16)(gSaveContext.save.time - 0x3FFC);
-    u16 temp;
-    u8 sp23 = ENBABA_GET_3F00(&this->actor);
+s32 EnBaba_InitTimePath(EnBaba* this, GlobalContext* globalCtx, ScheduleResult* scheduleResult) {
+    u16 now = SCHEDULE_TIME_NOW;
+    u16 startTime;
+    u8 path = ENBABA_GET_PATH(&this->actor);
 
-    if (D_80BAA4A8[arg2->result] >= 0) {
-        this->unk_410 = SubS_GetAdditionalPath(globalCtx, sp23, D_80BAA4A8[arg2->result]);
+    if (sAdditionalTimePaths[scheduleResult->result] >= 0) {
+        this->timePath = SubS_GetAdditionalPath(globalCtx, path, sAdditionalTimePaths[scheduleResult->result]);
     }
 
-    if (this->unk_410 == NULL) {
+    if (this->timePath == NULL) {
         return false;
     }
 
-    if ((this->unk_434 != 0) && (this->unk_436 >= 0)) {
-        temp = sp26;
+    if ((this->scheduleResult != 0) && (this->timePathTimeSpeed >= 0)) {
+        startTime = now;
     } else {
-        temp = arg2->time0;
+        startTime = scheduleResult->time0;
     }
 
-    if (arg2->time1 < temp) {
-        this->unk_424 = (temp - arg2->time1) + 0xFFFF;
+    if (scheduleResult->time1 < startTime) {
+        this->timePathTotalTime = (startTime - scheduleResult->time1) + (DAY_LENGTH - 1);
     } else {
-        this->unk_424 = arg2->time1 - temp;
+        this->timePathTotalTime = scheduleResult->time1 - startTime;
     }
 
-    this->unk_430 = sp26 - temp;
-    temp = this->unk_410->count - 2;
-    this->unk_428 = this->unk_424 / temp;
-    this->unk_438 = 0;
-    this->unk_42C = (this->unk_430 / this->unk_428) + 2;
-    this->unk_43C = 0;
+    this->timePathElapsedTime = now - startTime;
+
+    startTime = this->timePath->count - 2; // startTime resused to store total waypoints
+    this->timePathWaypointTime = this->timePathTotalTime / startTime;
```

Comment:
```suggestion
    numPointsMinusTwo = startTime = this->timePath->count - 2;
    this->timePathWaypointTime = this->timePathTotalTime / numPointsMinusTwo;
```
This is dumb but clearer? You can introduce `u16 numPointsMinusTwo;` at the bottom of the stack without affecting stack, then, use this variable like this and just slide `startTime` in there, and you have a matching and clearer docs?

Reply:
Hmm, it also seems to work everywhere else, though not all actors that use this have been fully decomped. I think I would be fine with it, though I think `numWaypoints` would be better.